sound/soc/sof/ipc4-pcm.c

```diff
@@ -66,16 +66,66 @@ static int sof_ipc4_trigger_pipelines(struct snd_soc_component *component,
 	struct snd_soc_pcm_runtime *rtd = asoc_substream_to_rtd(substream);
 	struct snd_sof_pcm_stream_pipeline_list *pipeline_list;
 	struct ipc4_pipeline_set_state_data *data;
+	struct snd_soc_dapm_widget_list *list;
 	struct snd_sof_widget *pipe_widget;
+	struct snd_soc_dapm_widget *widget;
 	struct sof_ipc4_pipeline *pipeline;
+	struct snd_sof_widget *swidget;
 	struct snd_sof_pcm *spcm;
+	int num_widgets;
 	int ret;
 	int i, j;
 
 	spcm = snd_sof_find_spcm_dai(component, rtd);
 	if (!spcm)
 		return -EINVAL;
 
+        list = spcm->stream[substream->stream].list;
+        for_each_dapm_widgets(list, num_widgets, widget) {
+		swidget = widget->dobj.private;
+
+		if (!swidget)
+                        continue;
```

Comment:
can this happen, and if yes is this a feature or a bug?

Reply:
Not sure. This is part of my pedantic copying of the behavior before rebasing on top of latest topic/sof-dev, so  I copied the old loop in every detail from previous branch. In the end the problem was not in this function at all. In the next version this loop is completely rewriten.